### PR TITLE
feat(suite-native): modals renderer

### LINKED ISSUE #10736: Unify modals renderers and avoid composition
Abstract renderers in `App.tsx` to its own component to avoid endlessly wrapping App with new modals.

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/10736

## Screenshots:
https://github.com/trezor/trezor-suite/assets/36101761/9770813d-b94b-4f4a-8ef9-17921d2824e4

### CODE REVIEW

File: suite-native/app/src/App.tsx [hunk 1]
```diff
@@ -95,27 +90,13 @@ const AppComponent = () => {
...
-                </AuthenticatorProvider>
-            </FormatterProvider>
+        <FormatterProvider config={formattersConfig}>
+            <MessageSystemBannerRenderer />
+            <RootStackNavigator />
+            <ModalsRenderer />
             {/* NOTE: Rendered as last item so that it covers the whole app screen */}
             <FeatureMessageScreen />
-        </>
+        </FormatterProvider>
```
Comment (matejkriz) [inline suite-native/app/src/App.tsx:99]: note to myself to check: 

The order of renderers has changed from:
```
MessageSystemBanner
RootStack
Toast
Notifications
Passhprase
Alert
Snow
Biometrics
FeatureMessage
```

to
```
MessageSystemBanner
RootStack
Alert
Passphrase
Toast
Snow
Biometrics
Notifications // no more commented out!
FeatureMessage
```

File: suite-native/app/src/ModalsRenderer.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <>
+        <AlertRenderer />
+        <PassphraseModalRenderer />
+        <ToastRenderer />
+        <Snow />
+        <BiometricsModalRenderer />
+        {/* Notifications are disabled until the problem with after-import notifications flooding is solved. */}
+        {/* More here: https://github.com/trezor/trezor-suite/issues/7721  */}
+        {/* <NotificationRenderer> */}
+        <NotificationRenderer />
```
Comment (matejkriz) [inline suite-native/app/src/ModalsRenderer.tsx]: Is it on purpose that NotificationRenderer is no more commented out?

File: suite-native/app/src/App.tsx (hunk 1 above)
Comment (matejkriz) [reply suite-native/app/src/App.tsx:99]: looks good, I see no problem here

**Review by matejkriz — CHANGES_REQUESTED** [review]
Looks way better 😍 Just double check the NotificationsRenderer please 🙏

File: suite-native/app/src/ModalsRenderer.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/app/src/ModalsRenderer.tsx]: 🫢🫢🫢
Definitely not correct.

Will fix.

**Review by Nodonisko — APPROVED** [review]

File: suite-native/app/src/ModalsRenderer.tsx (hunk 2 above)
Comment (juriczech) [reply suite-native/app/src/ModalsRenderer.tsx]: https://github.com/trezor/trezor-suite/pull/10801/commits/4db4ecf300cce3fe63e97ee4aa2cb6b32cc05e34

**Review by matejkriz — APPROVED** [review]

**Pull request merged**